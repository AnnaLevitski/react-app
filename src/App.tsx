import Card from "./components/Card";
import Button from "./components/Button";
import CheckBox from "./components/CheckBox";
import Input from "./components/Input";
import ListGroup from "./components/ListGroup";
import { SetStateAction, useState } from "react";

const items = ["Neque", "Porro quisquam", "Est qui ", "Dolorem"];

const handleSelectItem = (items: string) => {
  console.log(items);
};

function App() {
  const [isDisabledButton, setIsDisabledButton] = useState(true);
  const [isShownCard, setIsShownCard] = useState(false);
  const [value, setValue] = useState("");
  const [tempValue, setTempValue] = useState("");
  const [isDisabledCheckbox, setIsDisabledCheckbox] = useState(true);
  const handleChange = () => {
    setIsDisabledButton((current) => !current);
  };

  const handleClick = () => {
    setValue(tempValue);
    if (!isShownCard) setIsShownCard(true);
  };
  const handleClose = () => {
    setIsShownCard(false);
  };

  const handleNewValue = (val: string) => {
    console.log("Main " + val);
    if (!tempValue) setIsDisabledCheckbox(false);
    setTempValue(val);
  };

  return (
    <div>
      <ListGroup
        heading="Lorem Ipsum"
        items={items}
        onSelectItem={handleSelectItem}
      />
      <div className="input-group mb-3">
        <CheckBox disabled={isDisabledCheckbox} onChange={handleChange} />
        <Input onChange={handleNewValue} value={tempValue}></Input>
      </div>
      <Button isDisabled={isDisabledButton} onClick={handleClick}>
        Send
      </Button>
      <div>
        {isShownCard && (
          <Card closeClick={handleClose}>
            <strong>Holy guacamole! </strong> {value}
          </Card>
        )}
      </div>
    </div>
  );
}
export default App;
