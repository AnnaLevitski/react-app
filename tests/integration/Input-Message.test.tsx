import {
  fireEvent,
  getByText,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import App from "../../src/App";
import userEvent from "@testing-library/user-event";

describe("Input-Message", () => {
  it("should render properly", async () => {
    render(<App />);

    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
    const checkbox = screen.getByRole("checkbox");
    expect(checkbox).toBeInTheDocument();
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();

    const user = userEvent.setup();

    expect(checkbox).toBeDisabled();
    expect(button).toBeDisabled();

    fireEvent.change(input, { target: { value: "test text" } });

    expect(checkbox).toBeEnabled();
    expect(button).toBeDisabled();

    user.click(checkbox);
    user.click(button);

    await waitFor(() => {
      expect(screen.getByText(/Holy/i)).toBeInTheDocument();
    });
  });

  it("should match snapshot", () => {
    const app = render(<App />);
    expect(app).toMatchSnapshot();
  });
});
