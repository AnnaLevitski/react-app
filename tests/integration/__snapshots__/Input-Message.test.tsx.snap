// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input-Message > should 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h1>
          Home
        </h1>
        <div
          class="col-4"
        >
          <div
            class="list-group"
            id="list-tab"
            role="tablist"
          >
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action active"
              data-bs-toggle="list"
              href="#list-Home"
              id="list-[object Object]"
              role="tab"
            >
              Home
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Profile"
              id="list-[object Object]"
              role="tab"
            >
              Profile
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Messages"
              id="list-[object Object]"
              role="tab"
            >
              Messages
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Settings"
              id="list-[object Object]"
              role="tab"
            >
              Settings
            </a>
          </div>
        </div>
        <div
          class="col-8"
        >
          <p>
            Some placeholder content in a paragraph relating to "
            Home
            ".
          </p>
        </div>
      </div>
      <div
        class="input-group mb-3"
      >
        <div
          class="input-group-text"
        >
          <input
            class="form-check-input mt-0"
            disabled=""
            id="flexCheckDisabled"
            type="checkbox"
            value=""
          />
        </div>
        <input
          class="form-control"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        type="button"
      >
        Send
      </button>
      <div />
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h1>
        Home
      </h1>
      <div
        class="col-4"
      >
        <div
          class="list-group"
          id="list-tab"
          role="tablist"
        >
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action active"
            data-bs-toggle="list"
            href="#list-Home"
            id="list-[object Object]"
            role="tab"
          >
            Home
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Profile"
            id="list-[object Object]"
            role="tab"
          >
            Profile
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Messages"
            id="list-[object Object]"
            role="tab"
          >
            Messages
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Settings"
            id="list-[object Object]"
            role="tab"
          >
            Settings
          </a>
        </div>
      </div>
      <div
        class="col-8"
      >
        <p>
          Some placeholder content in a paragraph relating to "
          Home
          ".
        </p>
      </div>
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-text"
      >
        <input
          class="form-check-input mt-0"
          disabled=""
          id="flexCheckDisabled"
          type="checkbox"
          value=""
        />
      </div>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      disabled=""
      type="button"
    >
      Send
    </button>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input-Message > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h1>
          Home
        </h1>
        <div
          class="col-4"
        >
          <div
            class="list-group"
            id="list-tab"
            role="tablist"
          >
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action active"
              data-bs-toggle="list"
              href="#list-Home"
              id="list-[object Object]"
              role="tab"
            >
              Home
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Profile"
              id="list-[object Object]"
              role="tab"
            >
              Profile
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Messages"
              id="list-[object Object]"
              role="tab"
            >
              Messages
            </a>
            <a
              aria-controls="list-home"
              class="list-group-item list-group-item-action"
              data-bs-toggle="list"
              href="#list-Settings"
              id="list-[object Object]"
              role="tab"
            >
              Settings
            </a>
          </div>
        </div>
        <div
          class="col-8"
        >
          <p>
            Some placeholder content in a paragraph relating to "
            Home
            ".
          </p>
        </div>
      </div>
      <div
        class="input-group mb-3"
      >
        <div
          class="input-group-text"
        >
          <input
            class="form-check-input mt-0"
            disabled=""
            id="flexCheckDisabled"
            type="checkbox"
            value=""
          />
        </div>
        <input
          class="form-control"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        type="button"
      >
        Send
      </button>
      <div />
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h1>
        Home
      </h1>
      <div
        class="col-4"
      >
        <div
          class="list-group"
          id="list-tab"
          role="tablist"
        >
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action active"
            data-bs-toggle="list"
            href="#list-Home"
            id="list-[object Object]"
            role="tab"
          >
            Home
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Profile"
            id="list-[object Object]"
            role="tab"
          >
            Profile
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Messages"
            id="list-[object Object]"
            role="tab"
          >
            Messages
          </a>
          <a
            aria-controls="list-home"
            class="list-group-item list-group-item-action"
            data-bs-toggle="list"
            href="#list-Settings"
            id="list-[object Object]"
            role="tab"
          >
            Settings
          </a>
        </div>
      </div>
      <div
        class="col-8"
      >
        <p>
          Some placeholder content in a paragraph relating to "
          Home
          ".
        </p>
      </div>
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-text"
      >
        <input
          class="form-check-input mt-0"
          disabled=""
          id="flexCheckDisabled"
          type="checkbox"
          value=""
        />
      </div>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      disabled=""
      type="button"
    >
      Send
    </button>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
